### PR TITLE
Added an option to replace xkb indicator names

### LINKED ISSUE #1558: Add ability to change xkeyboard indicator name
Is it possible to add the ability to switch the text displayed by xkeyboard when displaying an indicator (i.e. instead of "Caps Lock" put an icon)?

Thank you, Gilad

Edit:

Written a PR:
https://github.com/jaagr/polybar/pull/1559

The result looks like this with my config:
screenshot from 2018-12-26 17-57-04

### PR DESCRIPTION
**Pull request by Gilnaa**
TEAM EDIT: Fixes #1558 
Closes #1048 

**Changelog Notes:**
This deprecates `label-indicator` in favor of `label-indicator-on`

### CODE REVIEW

**Comment by patrick96** [comment]
That one was fast, thanks!

Almost a year ago there was a PR #1048 that also adds this feature, but we forgot to review it. If the author is still around, I will go with that submission if possible.
If not, would you be willing to extend their PR and make changes to it instead of this one or would you rather work to get this PR here merged?

**Comment by Gilnaa** [comment]
Sure why not

**Comment by patrick96** [comment]
Alright, I'll let you know if the author is still around

**Comment by patrick96** [comment]
Hey @Gilnaa seems like the original author of #1048 isn't around anymore. If you are still up for it, please pull their changes (and incorporate your own changes if you want to) and submit a new PR. I will then review it

**Comment by Gilnaa** [comment]
Sure. It'll probably only happen during the weekend

**Comment by patrick96** [comment]
No rush, we have a lot on our plate as well ;)

**Comment by Gilnaa** [comment]
@patrick96 Would you mind reviewing?
I changed my addition a bit to fit with @ozelis original submission.

**Comment by Gilnaa** [comment]
@patrick96 have you had any time to review this?

**Comment by patrick96** [comment]
Not yet. I have exams coming up.

**Comment by Gilnaa** [comment]
Oh, no worries. Good luck!

File: src/modules/xkeyboard.cpp [hunk 1]
```diff
@@ -17,28 +17,101 @@ namespace modules {
...
+        const auto& indicator_str = m_keyboard->indicator_name(it);
+        auto key_name = string_util::replace(string_util::lower(indicator_str), " "s, ""s);
+        const auto indicator_key_on = "label-indicator-on-"s + key_name;
+        const auto indicator_key_off = "label-indicator-off-"s + key_name;
+
+        if (m_conf.has(name(), indicator_key_on)) {
+          m_indicator_on_labels.emplace(it, load_optional_label(m_conf, name(), indicator_key_on));
+        }
+        if (m_conf.has(name(), indicator_key_off)) {
+          m_indicator_on_labels.emplace(it, load_optional_label(m_conf, name(), indicator_key_off));
```
[the remark this answers is not in the document]
Comment (Gilnaa) [reply src/modules/xkeyboard.cpp]: Whoops. Why not optional-label?

**Comment by Gilnaa** [comment]
@patrick96 Please review again :)

**Comment by patrick96** [comment]
Will do, thanks!

File: src/modules/xkeyboard.cpp [hunk 2]
```diff
@@ -12,33 +12,109 @@ POLYBAR_NS
...
+
+    // Create keyboard object
+    query_keyboard();
+
     // Load config values
     m_blacklist = m_conf.get_list(name(), "blacklist", {});
+    
+    // load layout icons
+    m_layout_icons = factory_util::shared<iconset>();
+    m_layout_icons->add(DEFAULT_LAYOUT_ICON, load_optional_label(m_conf, name(), DEFAULT_LAYOUT_ICON, "s"));
```
Comment (patrick96) [inline src/modules/xkeyboard.cpp]:
```suggestion
    m_layout_icons->add(DEFAULT_LAYOUT_ICON, load_optional_label(m_conf, name(), DEFAULT_LAYOUT_ICON, ""s));
```

**Comment by Gilnaa** [comment]
Whoops, fixed

**Pull request merged**